### PR TITLE
[CELEBORN-1191][FOLLOWUP] Migrate the release script from Maven to SBT

### PR DESCRIPTION
**Pull request by cfmcgrady**
### What changes were proposed in this pull request?

As title

### Why are the changes needed?

1. extract `RELEASE_VERSION` from version.sbt instead of pom.xml
2. enable sbt when making binary distribution package

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Pass GA

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

**Comment by cfmcgrady** [comment]
thanks, merging to main(v0.5.0)/branch-0.4(v0.4.0).

File: build/release/release.sh [hunk 1]
```diff
@@ -27,9 +27,7 @@ ASF_USERNAME=${ASF_USERNAME:?"ASF_USERNAME is required"}
 ASF_PASSWORD=${ASF_PASSWORD:?"ASF_PASSWORD is required"}
 RELEASE_RC_NO=${RELEASE_RC_NO:?"RELEASE_RC_NO is required, e.g. 0"}
 
-RELEASE_VERSION=$(grep '<project.version>.*</project.version>' "${PROJECT_DIR}/pom.xml" -o \
-                | head -n 1 \
-                | sed 's/<\/*project.version>//g')
+RELEASE_VERSION=$(awk -F'"' '/ThisBuild \/ version/ {print $2}' version.sbt)
```
Comment (pan3793) [inline build/release/release.sh:30]: was this tested on both macOS and Linux? there are some differences in sed/awk/grep commands

Comment (cfmcgrady) [reply build/release/release.sh:30]: both macOS and Linux work as expected

macOS

```
$ uname -a
Darwin chenfudeMacBook-Pro-2.local 19.5.0 Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64
$ awk -F'"' '/ThisBuild \/ version/ {print $2}' version.sbt
0.4.1-SNAPSHOT
```

Linux
```
$ uname -a
Linux pop-os 6.2.6-76060206-generic #202303130630~1683753207~22.04~77c1465 SMP PREEMPT_DYNAMIC Wed M x86_64 x86_64 x86_64 GNU/Linux
$ cat version.sbt | grep -i version
 * The ASF licenses this file to You under the Apache License, Version 2.0
ThisBuild / version := "0.4.0"
$ awk -F'"' '/ThisBuild \/ version/ {print $2}' version.sbt
0.4.0
```

Comment (pan3793) [reply build/release/release.sh:30]: thanks for checking

**Pull request closed without merging**